### PR TITLE
Move to Meson 0.61

### PR DESCRIPTION
**Pull request by tristan957**
Meson 0.61 comes with install_symlink() which allows us to drop custom
code.

Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: meson.build
Comment (tristan957) [reply meson.build:14]: hse-java is moving to 0.62.

I am just using a venv for now, which is what we have done in the past.

`python3 -m venv /opt/venv; source /opt/venv/bin/activate; python3 -m pip install meson`

Comment (beckerg) [reply meson.build:14]: can we have that set up by cobbler?  it's probably a non-starter for most folk to require software that even the most bleeding edge distro doesn't support (IMO it's a PITA).

Comment (tristan957) [reply meson.build:14]: @bhaveshvasandani would know whether it is doable or not. Arch already has it :P. https://archlinux.org/packages/extra/any/meson/.

Fedora is a PITA because it doesn't update Meson at all.

**Comment by tristan957** [comment]
Note to self. Don't merge this one until: https://github.com/hse-project/hse/pull/237

**Review by alexttx — APPROVED** [review]

**Comment by bhaveshvasandani** [comment]
Now that we are on 0.62.0, we dont need this PR?

**Comment by tristan957** [comment]
0.61 is the lowest version which could build HSE. 0.62 is only necessary for hse-java.

**Comment by tristan957** [comment]
CI is now green. I think when Fedora 36 releases and we have access in our build machines, we should go ahead and merge this PR unless anyone has any objections.

**Review by bhaveshvasandani — APPROVED** [review]

**Pull request merged**
